backend/src/routes/auth.ts

```diff
@@ -160,4 +165,86 @@ export const authRoute = new Elysia({ prefix: '/auth' })
             status: 'success',
             message: `${user.email} has been verified`,
         };
+    })
+    .post('/sign-in', async ({ request, cookie }) => {
+        const formData = await request.formData();
+
+        const validEmaillPass = signInFormSchema.safeParse({
+            email: formData.get('email'),
+            password: formData.get('password'),
+        });
+
+        if (!validEmaillPass.success) {
+            return {
+                status: '400',
+                errors: validEmaillPass.error.flatten().fieldErrors,
+            };
+        }
+
+        const { email, password } = validEmaillPass.data;
+
+        let user_id = '';
+
+        try {
+            const queryAuthUserData = await sql`
+            SELECT id, hashed_password
+            FROM auth_user
+            WHERE email = ${email}
+            `;
+
+            const queriedHashedPassword = queryAuthUserData[0].hashed_password;
+            user_id = queryAuthUserData[0].id;
+
+            const isPasswordMatch = await new Argon2id().verify(
+                queriedHashedPassword,
+                password,
+            );
+
+            if (!isPasswordMatch) {
+                return {
+                    status: '401',
+                    message: 'email or password is incorrect',
+                };
+            }
+
+            if (queriedHashedPassword.length == 0) {
```

Comment:
This can be used to attack by finding existing, and non-existing account.

Reply:
remove